codec-redis/src/main/java/io/netty/handler/codec/redis/ArrayRedisMessage.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.util.AbstractReferenceCounted;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.StringUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Arrays of <a href="http://redis.io/topics/protocol">RESP</a>.
+ */
+public class ArrayRedisMessage extends AbstractReferenceCounted implements RedisMessage {
+
+    private final List<RedisMessage> children;
+
+    private ArrayRedisMessage() {
+        children = Collections.emptyList();
+    }
+
+    /**
+     * Creates a {@link ArrayRedisMessage} for the given {@code content}.
+     *
+     * @param children the children.
+     */
+    public ArrayRedisMessage(List<RedisMessage> children) {
+        // do not retain here. children are already retained when created.
+        this.children = ObjectUtil.checkNotNull(children, "children");
+    }
+
+    /**
+     * Get children of this Arrays. It can be null or empty.
+     *
+     * @return list of {@link RedisMessage}s.
+     */
+    public List<RedisMessage> children() {
+        return children;
+    }
+
+    /**
+     * Returns whether the content of this message is {@code null}.
+     *
+     * @return indicates whether the content of this message is {@code null}.
+     */
+    public boolean isNull() {
```

Comment:
@Scottmitch , It's nice for encoder/decoder codes. But, I agree with @mp911de . Even if `null bulk string` can be an alternative for `null array` in RESP (http://redis.io/topics/protocol#resp-arrays), latest Redis server source code uses 2 diffrent types (`shared.nullbulk`, `shared.nullmultibulk`) for it. https://github.com/antirez/redis/blob/3.2/src/server.c#L1351

And, basically, this is encoder/decoder. If we merge 2 nulls (null bulk string, null array) to only one null, it breaks symmetry of Redis encoder/decoder. I think we have to keep follow cases. Then, users can make Redis proxy using this without concerns of changing data.

* `$-1\r\n` =[encode]=> null bulk string object
* null bulk string object =[decode]=> `$-1\r\n`
* `*-1\r\n` =[encode]=> null array object
* null array object =[decode]=> `*-1\r\n`